### ISSUE TITLE
Type mismatch

### ISSUE THREAD

**Issue by plantain-00** [Microsoft/TypeScript#8590]
**TypeScript Version:** 

1.8.10

**Code**

```ts
function f(a: { b?: string; c?: number; }){
	// something else
}
f({b: ""}); //  no error as intended
f({b: 1}); //  error as intended
f(1); // expected: error, actually: no error
f(""); // expected: error, actually: no error
```

**Expected behavior:**
see comments in the code.
**Actual behavior:** 
see comments in the code.

**Comment by malibuzios** [comment]
TypeScript currently treats all-optional types similarly to `{}` types (which are almost as weak as `any` in practice):

There have been past attempts to address this:

* One pull request dating back to July 2015, which was eventually closed: #3842
* Newer issue which is now labeled as 'Accepting PRs' with milestone of 'Community': #7485

**Comment by plantain-00** [comment]
Seems duplicate.
I searched as a key word "Type mismatch", but didn't find those.
I will close this issue, and focus on #7485.
